src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,41 @@
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+public class Contact {
+    
+    public final String value;
+    private boolean isPrivate;
+    
+    public Contact(String value, boolean isPrivate, String validationRegex, String errorMessage) throws IllegalValueException{
+        if(!isValid(value, validationRegex))
+            throw new IllegalValueException(errorMessage);
+        this.value = value;
+        this.isPrivate = isPrivate; 
+    }
+    
+    @Override
+    public String toString() {
+        return value;
+    }
+    
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof Contact // instanceof handles nulls
+                && this.value.equals(((Contact) other).value)); // state check
```

Comment:
Note that now an Email and Address having the same value will be considered equal even if they are of different types.